File: app/src/main/java/com/vivekvishwanath/androidanimatedimages/MainActivity.java
package com.vivekvishwanath.androidanimatedimages;

import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.Spinner;

public class MainActivity extends AppCompatActivity {

    Spinner spinner;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        FragmentManager fragmentManager = getSupportFragmentManager();
        final FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();

        spinner = findViewById(R.id.spinner);
        spinner.setOnItemSelectedListener(new Spinner.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                switch (position) {
                    case 0:
                        getSupportFragmentManager().beginTransaction()
                                .replace(R.id.fragment_container, new MonkeyFragment())
                                .addToBackStack(null)
                                .commit();
                        break;
                    case 1:
                        getSupportFragmentManager().beginTransaction()
                                .replace(R.id.fragment_container, new ElonMuskFragment())
                                .addToBackStack(null)
                                .commit();
                        break;
                }
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {

            }
        });
    }
}

File: app/src/main/java/com/example/androidanimatedimages/acitivities/MainActivity.kt
package com.example.androidanimatedimages.acitivities

import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.androidanimatedimages.R
import com.example.androidanimatedimages.fragments.AnimatedDrawableFragment
import com.example.androidanimatedimages.fragments.AnimatedVectorFragment
import com.example.androidanimatedimages.fragments.GifFragment

class MainActivity : AppCompatActivity(), AnimatedDrawableFragment.OnFragmentInteractionListener,
    AnimatedVectorFragment.OnFragmentInteractionListener,
    GifFragment.OnFragmentInteractionListener {
    override fun onFragmentInteraction(uri: Uri) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    //i could probably make just one fragment that would determine which kind  of drawable is being passed to it, but we're gonna
    //do it this way


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val gifFragment = GifFragment()
        val animatedDrawableFragment = AnimatedDrawableFragment()
        val animatedVectorFragment = AnimatedVectorFragment()

    /*
            if we need a bundle...
    val bundle = Bundle()
        bundle.putString(ITEM_KEY, item.id)

        //fragment.arguments = bundle
        fragment.setArguments(bundle)
*/


            supportFragmentManager.beginTransaction()
                .replace(R.id.gif_frame, gifFragment)
                .replace(R.id.animated_drawable_frame, animatedDrawableFragment)
                .replace(R.id.animated_vector_frame, animatedVectorFragment)
                .commit()


    }
}
